### PR TITLE
Blog post introducing kwok

### PR DESCRIPTION
**Pull request by wzshiming**
mirror from https://github.com/kubernetes/website/pull/39653

### CODE REVIEW

File: content/en/blog/2023/2023-03-01-introducing-kwok/index.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+---
+layout: blog
+title: "Introducing KWOK: Kubernetes WithOut Kubelet"
+date: 2023-03-01
+slug: introducing-kwok
+---
+
+**Author:** Shiming Zhang (DaoCloud), Wei Huang (Apple), Yibo Zhuang (Apple)
+
+<img align="right" width="300em" src="/blog/2023/03/01/introducing-kwok/kwok.svg"/>
```
Comment (sftim) [inline content/en/blog/2023/2023-03-01-introducing-kwok/index.md:10]: FYI 300em is **very** big (around 1200mm). Try 15em?

File: content/en/blog/2023/2023-03-01-introducing-kwok/index.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+- **Functionality**: KWOK is not a kubelet and many behaviors may be different, such as pod lifecycle management, volume mounting, device plugins, etc. It only simulates updates of node status and pod status.
+- **Accuracy**: KWOK does not reflect the actual performance or behavior of real nodes under different workloads or environments. It only approximates some behaviors based on simple formulas.
+- **Security**: KWOK does not enforce any security policies or mechanisms on simulated nodes. It assumes that all requests from the kube-apiserver are authorized and valid.
+
+## Getting Started
+
+If you are interested in trying out KWOK, please check its [documents] for more details.
+
+{{< figure src="/blog/2023/03/01/introducing-kwok/manage-clusters.svg" alt="KWOK Manage Clusters" caption="KWOK Manage Clusters" >}}
```
Comment (sftim) [inline content/en/blog/2023/2023-03-01-introducing-kwok/index.md:58]:
```suggestion
{{< figure src="/blog/2023/03/01/introducing-kwok/manage-clusters.svg" alt="Animation of a terminal showing KWOW in use" caption="Using KWOK to manage simulated clusters" >}}
```

File: content/en/blog/2023/2023-03-01-introducing-kwok/index.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+## What are the limitations?
+
+KWOK is not intended to replace others completely. It has some limitations that you should be aware of:
+
+- **Functionality**: KWOK is not a kubelet and many behaviors may be different, such as pod lifecycle management, volume mounting, device plugins, etc. It only simulates updates of node status and pod status.
+- **Accuracy**: KWOK does not reflect the actual performance or behavior of real nodes under different workloads or environments. It only approximates some behaviors based on simple formulas.
+- **Security**: KWOK does not enforce any security policies or mechanisms on simulated nodes. It assumes that all requests from the kube-apiserver are authorized and valid.
+
+## Getting Started
```
Comment (sftim) [inline content/en/blog/2023/2023-03-01-introducing-kwok/index.md:54]:
```suggestion
## Getting started
```

File: content/en/blog/2023/2023-03-01-introducing-kwok/index.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+<img align="right" width="300em" src="/blog/2023/03/01/introducing-kwok/kwok.svg"/>
+
+Have you ever wondered how to set up a cluster of thousands of nodes just in seconds, how to simulate real nodes with a low resource footprint, and how to test your Kubernetes controller at scale without spending much on infrastructure?
+
+If you say "yes" to any of these questions, then you might be interested in KWOK, a toolkit that enables you to create a cluster of thousands of nodes in seconds.
+
+## What is KWOK?
+
+KWOK stands for Kubernetes WithOut Kubelet. So far, we provide two tools:
```
Comment (sftim) [inline content/en/blog/2023/2023-03-01-introducing-kwok/index.md:18]:
```suggestion
KWOK stands for Kubernetes WithOut Kubelet. So far, we (Kubernetes SIG Scheduling) provide two tools:
```

File: content/en/blog/2023/2023-03-01-introducing-kwok/index.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+- **kwok**: The core of this project. It simulates thousands of fake nodes.
+- **kwokctl**: A CLI to facilitate the creation and management of clusters simulated by **kwok**.
```
Comment (sftim) [inline content/en/blog/2023/2023-03-01-introducing-kwok/index.md:21]:
```suggestion
`kwok`
: The core of this project. The `kwok` tool simulates thousands of fake nodes.

`kwokctl`
: A CLI tool to facilitate the creation and management of clusters with nodes simulated by `kwok`.
```

File: content/en/blog/2023/2023-03-01-introducing-kwok/index.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+slug: introducing-kwok
+---
+
+**Author:** Shiming Zhang (DaoCloud), Wei Huang (Apple), Yibo Zhuang (Apple)
+
+<img align="right" width="300em" src="/blog/2023/03/01/introducing-kwok/kwok.svg"/>
+
+Have you ever wondered how to set up a cluster of thousands of nodes just in seconds, how to simulate real nodes with a low resource footprint, and how to test your Kubernetes controller at scale without spending much on infrastructure?
+
+If you say "yes" to any of these questions, then you might be interested in KWOK, a toolkit that enables you to create a cluster of thousands of nodes in seconds.
```
Comment (sftim) [inline content/en/blog/2023/2023-03-01-introducing-kwok/index.md:14]: (wording nit; optional to accept)
```suggestion
If you answered "yes" to any of these questions, then you might be interested in KWOK, a toolkit that enables you to create a cluster of thousands of nodes in seconds.
```

File: content/en/blog/2023/2023-03-01-introducing-kwok/index.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+- **kwok**: The core of this project. It simulates thousands of fake nodes.
+- **kwokctl**: A CLI to facilitate the creation and management of clusters simulated by **kwok**.
+
+## Why use KWOK?
+
+KWOK has several advantages:
+
+- **Speed**: You can create and delete clusters and nodes almost instantly, without waiting for boot or provisioning.
+- **Compatibility**: KWOK works with any tools or clients that are compliant with Kubernetes APIs, such as kubectl, helm, kui, etc.
+- **Portability**: KWOK has no special hardware or software requirements. You can run it on your laptop, desktop or any cloud instance that supports Docker. You can also run it alongside other Kubernetes clusters or tools.
```
Comment (sftim) [inline content/en/blog/2023/2023-03-01-introducing-kwok/index.md:29]: Is Docker required (could you also use Podman)? If there's no hard dependency on Docker, can we reword to make that clear?

File: content/en/blog/2023/2023-03-01-introducing-kwok/index.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+If you are interested in trying out KWOK, please check its [documents] for more details.
+
+{{< figure src="/blog/2023/03/01/introducing-kwok/manage-clusters.svg" alt="KWOK Manage Clusters" caption="KWOK Manage Clusters" >}}
+
+## Getting Involved
+
+If you're interested in future development and discussions on KWOK, you can get involved in SIG Scheduling through the following ways:
+
+- Slack: [#kwok], [#kwok-dev], [#sig-scheduling]
```
Comment (sftim) [inline content/en/blog/2023/2023-03-01-introducing-kwok/index.md:64]:
```suggestion
- Slack: [#kwok], [#kwok-dev], [#sig-scheduling] (visit https://slack.k8s.io/ for a workspace invitation)
```

File: content/en/blog/2023/2023-03-01-introducing-kwok/index.md (hunk 1 above)
Comment (sftim) [reply content/en/blog/2023/2023-03-01-introducing-kwok/index.md:10]:
```suggestion
<img style="float: right; display: inline-block; margin-left: 2em; max-width: 15em;" src="/blog/2023/03/01/introducing-kwok/kwok.svg" alt="KWOK logo" />
```
(untested)

**Review by sftim — COMMENTED** [review]
The image size fix feels important. Rest of the article LGTM; fixups and tweaks are still welcome.

File: content/en/blog/2023/2023-03-01-introducing-kwok/index.md (hunk 7 above)
Comment (wzshiming) [reply content/en/blog/2023/2023-03-01-introducing-kwok/index.md]: Updated, in addition to Docker, Nerdctl and binary are supported and Podman may be supported in the future.

**Comment by sftim** [comment]
/retitle [WIP] Add blog post for introducing kwok

Informally LGTM (I'll try to take more of a look later)

**Comment by Huang-Wei** [comment]
@wzshiming I haven't compared the content with the k/website one thoroughly. As long as the mirrored version resolved all the comments, I'm fine with this PR as well.

**Comment by wzshiming** [comment]
Yes, both are the same.

**Comment by sftim** [comment]
Is this still a work in progress? What needs to happen before it's ready to review?

**Comment by Huang-Wei** [comment]
> Is this still a work in progress? What needs to happen before it's ready to review?

It's ready for reviewing.

/retitle Blog post introducing kwok

File: content/en/blog/2023/2023-03-01-introducing-kwok/index.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
+---
+layout: blog
+title: "Introducing KWOK: Kubernetes WithOut Kubelet"
+date: 2023-03-01
+slug: introducing-kwok
+---
+
+**Author:** Shiming Zhang (DaoCloud), Wei Huang (Apple), Yibo Zhuang (Apple)
```
Comment (sftim) [inline content/en/blog/2023/2023-03-01-introducing-kwok/index.md:8]:
```suggestion
**Authors:** Shiming Zhang (DaoCloud), Wei Huang (Apple), Yibo Zhuang (Apple)
```

**Review by sftim — COMMENTED** [review]
Thanks

The filename should actually be `content/en/blog/2023/introducing-kwok/index.md`

I'll send in a PR to fix that.

**Review by sftim — COMMENTED** [review]
/hold
/lgtm
/approve

**Pull request merged**